maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -72,6 +78,8 @@
     private final LifecycleDependencyResolver lifeCycleDependencyResolver;
     private final ExecutionEventCatapult eventCatapult;
 
+    private final ReadWriteLock aggregatorLock = new ReentrantReadWriteLock( true );
```

Comment:
Does it really have to be fair?

Reply:
No idea, I just reused the code from the other PR. You think it should be non-fair ?